libraries/SDFS/src/SDFS.h

```diff
@@ -113,8 +113,22 @@ class SDFSImpl : public FSImpl
         info.blockSize = _fs.vol()->blocksPerCluster() * 512;
         info.pageSize = 0; // TODO ?
         info.maxPathLength = 255; // TODO ?
-        info.totalBytes =_fs.vol()->volumeBlockCount() * 512;
-        info.usedBytes = info.totalBytes - (_fs.vol()->freeClusterCount() * _fs.vol()->blocksPerCluster() * 512);
+        info.totalBytes =_fs.vol()->volumeBlockCount() * 512LL;
+        info.usedBytes = info.totalBytes - (_fs.vol()->freeClusterCount() * _fs.vol()->blocksPerCluster() * 512LL);
+        return true;
+    }
+
+    bool info(FSInfo& info) override {
+        FSInfo64 i;
+        if (!info64(i)) {
+            return false;
+        }
+        info.blockSize     = i.blockSize;
+        info.pageSize      = i.pageSize;
+        info.maxOpenFiles  = i.maxOpenFiles;
+        info.maxPathLength = i.maxPathLength;
+        info.totalBytes    = (size_t)i.totalBytes;
```

Comment:
I would assert here if the uint64 value can't be contained in 32bits, with a medsage that the sd card capacity is larger than supported. 
Alternatively, at least print out some debug msg or something if the is encountered.

Reply:
assert would break backward compatibility (the bug was already there)
What about a print (in no debug mode) and return max(32bits) ?

Reply:
Done in latest push.